e2e/customized-env/customized_env_test.go

```diff
@@ -0,0 +1,265 @@
+package customized_env_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/aws/copilot-cli/e2e/internal/client"
+	"github.com/aws/copilot-cli/e2e/internal/command"
+)
+
+var _ = Describe("Customized Env", func() {
+	Context("when creating a new app", func() {
+		var appInitErr error
+		BeforeAll(func() {
+			_, appInitErr = cli.AppInit(&client.AppInitRequest{
+				AppName: appName,
+				Tags: map[string]string{
+					"e2e-test": "customized-env",
+				},
+			})
+		})
+
+		It("app init succeeds", func() {
+			Expect(appInitErr).NotTo(HaveOccurred())
+		})
+
+		It("app init creates a copilot directory", func() {
+			Expect("./copilot").Should(BeADirectory())
+		})
+
+		It("app ls includes new app", func() {
+			apps, err := cli.AppList()
+			Expect(err).NotTo(HaveOccurred())
+			Expect(apps).To(ContainSubstring(appName))
+		})
+
+		It("app show includes app name", func() {
+			appShowOutput, err := cli.AppShow(appName)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(appShowOutput.Name).To(Equal(appName))
+			Expect(appShowOutput.URI).To(BeEmpty())
+		})
+	})
+
+	Context("when deploying resources to be imported", func() {
+		It("deploy vpc cloudformation stack", func() {
+			err := command.Run("aws", []string{"cloudformation", "create-stack", "--stack-name",
+				vpcStackName, "--template-body", vpcStackTemplatePath})
+			Expect(err).NotTo(HaveOccurred(), "create vpc cloudformation stack")
+			err = command.Run("aws", []string{"cloudformation", "wait", "stack-create-complete", "--stack-name", vpcStackName})
+			Expect(err).NotTo(HaveOccurred(), "vpc stack create complete")
+		})
+		It("parse vpc stack output", func() {
+			var b bytes.Buffer
+			err := command.Run("bash", []string{"-c", fmt.Sprintf("aws cloudformation describe-stacks --stack-name %s | jq -r .Stacks[0].Outputs", vpcStackName)}, command.Stdout(&b))
+			Expect(err).NotTo(HaveOccurred(), "describe vpc cloudformation stack")
```

Comment:
Maybe we can throw this into a aws.go file? (Like how we have the CLI one). The Output can be a VPC object or whatever - that way the tests stay really focused on the expected output.